### PR TITLE
feat(honeycomb): Add support for secrets provider

### PR DESCRIPTION
**Pull request by martin308**
## Which problem is this PR solving?

Using an external secrets provider.

## Short description of the changes

Adds ability to specify additional volume mounts for the agents container and additional volumes for the pod spec. Also allows specification of the key used for the secret.

### CODE REVIEW

**Review by JamieDanielson — COMMENTED** [review]
One note about being more explicit in the comment describing the secret key, and a question about whether we need to specify the default. Otherwise this LGTM, would like another set of eyes from another teammate who knows helm a bit better though.

File: charts/honeycomb/values.yaml
Comment (martin308) [reply charts/honeycomb/values.yaml:9]: I believe this would just be a stylistic thing, so happy to do whichever way y'all would like to have it done. Just let me know can I can update

**Comment by martin308** [comment]
@JamieDanielson thanks for the review, re-requested @honeycombio/pipeline-team to get another review

**Review by codeboten — COMMENTED** [review]
The change looks good. An update to https://github.com/honeycombio/helm-charts/tree/main/charts/honeycomb#parameters is needed as well

**Comment by martin308** [comment]
@codeboten how's this look? [`c0a0985`](https://github.com/honeycombio/helm-charts/pull/356/commits/c0a0985a4322daffade151b87cdab8c7e5970029)

File: charts/honeycomb/README.md
Comment (codeboten) [inline charts/honeycomb/README.md:83]: There's also a reference to the `api-key` field in line 71, not sure if needs updating as well

Comment (martin308) [reply charts/honeycomb/README.md:83]: ah good catch, updating

**Review by codeboten — APPROVED** [review]

File: charts/honeycomb/values.yaml
Comment (JamieDanielson) [reply charts/honeycomb/values.yaml:9]: Hmm I guess it would be more self-documenting, I'm inclined to add it unless there's a specific reason not to

Comment (martin308) [reply charts/honeycomb/values.yaml:9]: so add it to the values file and remove the usage of the `default` function?

**Review by JamieDanielson — APPROVED** [review]

**Pull request merged**
